compiler/daml-extension/src/webview.js

```diff
@@ -3,17 +3,27 @@
 
 const vscode = acquireVsCodeApi();
 function show_archived_changed() {
-  document.body.classList.toggle('hide_archived', !document.getElementById('show_archived').checked);
+  const isChecked = document.getElementById('show_archived').checked;
+  document.body.classList.toggle('hide_archived', !isChecked);
+  vscode.postMessage({
+    'command': 'set_show_archived',
+    'value': isChecked
+  });
 }
 function toggle_detailed_disclosure() {
-  document.body.classList.toggle('hidden_disclosure', !document.getElementById('show_detailed_disclosure').checked);
+  const isChecked = document.getElementById('show_detailed_disclosure').checked;
+  document.body.classList.toggle('hidden_disclosure', !isChecked);
+  vscode.postMessage({
+    'command': 'set_show_detailed_disclosure',
+    'value': isChecked
+  });
```

Comment:
These two can also be factored into one.

Reply:
factored out.